README.md

```diff
@@ -44,3 +44,22 @@ To test ([Docker](https://docs.docker.com/engine/install) is required) run:
 ```bash
 npm run test
 ```
+
+## Implementation and Interface Notes
+
+### Temporal Types
+
+Suitable JS type to store Memgrpah temporal types don't exist. In particular,
+it's impossible to convert `mg_duration` and `mg_local_time` to the `Date`
+type. Since [the temporal
+specification](https://github.com/tc39/proposal-temporal) is not yet widely
+supported, the decision was to expose plain JS objects (dict) with the exact
+fields `mgclient` is providing. In addition, when possible (`mg_date` and
+`mg_local_date_time`), are converted into objects which have `date` property,
+which in fact, is the `Date` representation of these types. Keep in mind the
+loss of precision because `Date` can only store millisecond precision, but the
```

Comment:
This is slightly misleading, because `Date` is actually `Datetime` in JS. Maybe reword this to:

JS `Date` time fields can only store up to `milliseconds` precision. However,  Memgraph supports microsecond precision for the local time and therefore any use of this class can potentially cause loss of information.

Reply:
Better now?